### PR TITLE
Add metrics for created operands

### PR DESCRIPTION
**Pull request by rubenvp8510**
Fixes https://github.com/grafana/tempo-operator/issues/838

This PR adds two metrics to the operator prometheus endpoints, which indicates how many instances per storage are u sed, and how many instances are managed/unmanaged.

### CODE REVIEW

File: internal/crdmetrics/tempo_stack_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	}
+}
+
+func TestValueObservedMetrics(t *testing.T) {
+	s := scheme.Scheme
+
+	// Add jaeger to schema
+	s.AddKnownTypes(v1alpha1.GroupVersion, &v1alpha1.TempoStack{}, &v1alpha1.TempoStackList{})
+
+	// Create jaeger instances
```
Comment (iblancasa) [inline internal/crdmetrics/tempo_stack_test.go:102]:
```suggestion
	// Create tempo instances
```

File: internal/crdmetrics/tempo_stack_test.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		&tempoUnmanaged,
+		&tempoUnmanaged2,
+	}
+	expected := []expectedMetric{
+		newExpectedMetric(managedMetric, attribute.String("state", string(v1alpha1.ManagementStateManaged)), 3),
+		newExpectedMetric(managedMetric, attribute.String("state", string(v1alpha1.ManagementStateUnmanaged)), 2),
+		newExpectedMetric(storageBackendMetric, attribute.String("type", string(v1alpha1.ObjectStorageSecretGCS)), 1),
+		newExpectedMetric(storageBackendMetric, attribute.String("type", string(v1alpha1.ObjectStorageSecretS3)), 2),
+		newExpectedMetric(storageBackendMetric, attribute.String("type", string(v1alpha1.ObjectStorageSecretAzure)), 2),
+	}
```
Comment (iblancasa) [inline internal/crdmetrics/tempo_stack_test.go:141]: Wouldn't it make sense to do this assertions in a loop?

Comment (rubenvp8510) [reply internal/crdmetrics/tempo_stack_test.go:141]: At the end this is evaluated in a loop, but this way it makes me easier to see which metrics I'm evaluating. I just fin easier to read.

**Review by iblancasa — APPROVED** [review]

File: internal/crdmetrics/tempo_stack_test.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/grafana/tempo-operator/apis/tempo/v1alpha1"
+)
+
+const (
+	AgentSideCar   = "Sidecar"
```
Comment (andreasgerstmayr) [inline internal/crdmetrics/tempo_stack_test.go:23]: Is this used anywhere?

Comment (rubenvp8510) [reply internal/crdmetrics/tempo_stack_test.go:23]: Actually no. I'll remove it.

**Review by andreasgerstmayr — COMMENTED** [review]
LGTM :+1:

Could we add the new metrics to the existing `monitoring` e2e test?
I think we also need to set the `openshift.io/cluster-monitoring: "true"` label on the namespace (in the e2e test), but except that, only the `check_metrics.sh` script needs to be update to add the new metric names afaics.

**Review by andreasgerstmayr — APPROVED** [review]

**Comment by rubenvp8510** [comment]
> LGTM 👍
> 
> Could we add the new metrics to the existing `monitoring` e2e test? I think we also need to set the `openshift.io/cluster-monitoring: "true"` label on the namespace (in the e2e test), but except that, only the `check_metrics.sh` script needs to be update to add the new metric names afaics.

I'll address this in a separate PR :) but yes, it worth to add it to our e2e.

**Pull request merged**